### PR TITLE
[BAEL 3739] -  Workarounds for String contains case insensitive in Java

### PR DESCRIPTION
**Pull request by carloscaverobarca**
## Proposed Changes

  - Add the JMH dependencies
  - Shade the jar to avoid _ERROR: Unable to find the resource: /META-INF/BenchmarkList_
   - Add README file

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests
- [x] Build locally
- [x] Code format / lint
- [x] Docs have been reviewed and added

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by carloscaverobarca** [comment]
@glmartin I have included the Unit Tests. Please, could you check if everything is ok right now? I am finishing the article. I will inform you through JIRA when it is finished.

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * BAEL-3739: Different ways to solve the contains() case insensitive behavior.
+ */
+public class CaseInsensitiveWorkaroundsUnitTest {
+
+    private String src = "Lorem ipsum dolor sit amet";
+    private String dest = "lorem";
+
+    @Test
+    public void givenString_whenCallingContainsWithToLowerOrUpperCase_shouldReturnTrue() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]:
```suggestion
    public void givenString_whenCallingContainsWithToLowerOrUpperCase_thenReturnsTrue() {
```

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        // Use toLowerCase to avoid case insensitive issues
+        Assert.assertTrue(src.toLowerCase().contains(dest.toLowerCase()));
+
+        // Use toUpperCase to avoid case insensitive issues
+        Assert.assertTrue(src.toUpperCase().contains(dest.toUpperCase()));
+    }
+    
+    @Test
+    public void givenString_whenCallingStringMatches_shouldReturnTrue() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]:
```suggestion
    public void givenString_whenCallingStringMatches_thenReturnsTrue() {
```

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        // Use String Matches to avoid case insensitive issues
+        Assert.assertTrue(src.matches("(?i).*" + dest + ".*"));
+    }
+    
+    @Test
+    public void givenString_whenCallingStringRegionMatches_shouldReturnTrue() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]:
```suggestion
    public void givenString_whenCallingStringRegionMatches_thenReturnsTrue() {
```

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        // Use String Region Matches to avoid case insensitive issues
+        CaseInsensitiveWorkarounds comparator = new CaseInsensitiveWorkarounds();
+        Assert.assertTrue(comparator.processRegionMatches(src, dest));
+    }
+    
+    
+    @Test
+    public void givenString_whenCallingPaternCompileMatcherFind_shouldReturnTrue() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]:
```suggestion
    public void givenString_whenCallingPaternCompileMatcherFind_thenReturnsTrue() {
```

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        // Use Pattern Compile Matcher and Find to avoid case insensitive issues
+        Assert.assertTrue(Pattern.compile(Pattern.quote(dest), 
+            Pattern.CASE_INSENSITIVE) .matcher(src) .find());
+    }
+    
+    @Test
+    public void givenString_whenCallingStringUtilsContainsIgnoreCase_shouldReturnTrue() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]:
```suggestion
    public void givenString_whenCallingStringUtilsContainsIgnoreCase_thenReturnsTrue() {
```

File: core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java (hunk 1 above)
Comment (carloscaverobarca) [reply core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/contains/CaseInsensitiveWorkaroundsUnitTest.java]: Totally agree with, I accepted the proposed changes

**Pull request merged**